### PR TITLE
Fix missing parentheses of inorder print in recursive style

### PR DESCRIPTION
**Pull request by FissoreD**
The returned inorder expression only has usefull parenthesis, or every one if needed. 

❗ The formatter has moved the inner class TraversalEdge to the bottm of the class TreeTraversal.

### CODE REVIEW

File: src/main/java/cryptator/tree/TreeTraversals.java [hunk 1]
```diff
@@ -105,21 +77,96 @@ public static void postorderTraversal(final ICryptaNode root, final ITraversalNo
         }
     }
 
+    private static void buildDecoratedTree(final DecoratedTree decoratedNode){
```
Comment (arnaud-m) [inline src/main/java/cryptator/tree/TreeTraversals.java:80]: First _recursive_ pass that decorates the tree.

File: src/main/java/cryptator/tree/TreeTraversals.java [hunk 2]
```diff
@@ -105,21 +77,96 @@ public static void postorderTraversal(final ICryptaNode root, final ITraversalNo
...
+        if (decoratedNode.node.isInternalNode()){
+            final CryptaOperator op = decoratedNode.node.getOperator();
+            buildDecoratedTree(new DecoratedTree(op, decoratedNode, true));
+            buildDecoratedTree(new DecoratedTree(op, decoratedNode,false));
+        }
+    }
+
     public static void inorderTraversal(final ICryptaNode root, final ITraversalNodeConsumer traversalNodeConsumer) {
```
Comment (arnaud-m) [inline src/main/java/cryptator/tree/TreeTraversals.java:88]: Second _iterative_ pass on the decorated tree.

**Review by arnaud-m — CHANGES_REQUESTED** [review]
Well done identifying the conditions for the minimal form. 
 
The difficulty with the iterative approach is to put the right/closing parenthesis.
Since it is not a critical code and the size of the tree will remain small, we **fall back to a recursive approach**. 

- Restore the previous `inorderTraversal` method.
- Specialize the `writeInorder` method. 
  - It does not call the `inorderTraversal`.
  - It rather performs its own recursive traversal. 
  - It has an option to trigger the minimal/full parenthesized form.    

Do not puth a space before or after a parenthesis.

**Comment by FissoreD** [comment]
I've update the inorder print function as requested. 
The same function is overloaded and accepts a boolean for the full parenthesis mode.
The previous method inorder traversal has been restored as it was before my first modification.

File: src/main/java/cryptator/tree/TreeUtils.java [hunk 3]
```diff
@@ -54,27 +54,48 @@ public static String writePostorder(final ICryptaNode root) {
         return out.toString();
     }
 
-    public static void writeInorder(final ICryptaNode root, final OutputStream outstream) {
+    public static void writeInorder(final ICryptaNode node,
+                                    final PrintWriter out, int leftPar, int rightPar,
+                                    int putPar, final CryptaOperator oldOp, boolean isLeft,
+                                    final boolean allParenthesis){
```
Comment (arnaud-m) [inline src/main/java/cryptator/tree/TreeUtils.java:60]: Too many parameters.

**Review by arnaud-m — APPROVED** [review]

**Pull request merged**